src/Command/Redis/XINFO.php

```diff
@@ -47,17 +48,61 @@ private function setStreamArguments(array $arguments): void
     }
 
     public function parseResponse($data)
+    {
+        if ($this->getArgument(0) === 'STREAM') {
+            return $this->parseStreamResponse($data);
+        }
+
+        return $this->parseDict($data);
+    }
+
+    private function parseStreamResponse($data): array
+    {
+        $result = CommandUtility::arrayToDictionary($data, null, false);
+
+        if (isset($result['entries'])) {
+            $result['entries'] = $this->parseDict($result['entries']);
+        }
+
+        if (isset($result['groups']) && is_array($result['groups'])) {
```

Comment:
The logic of parsing is quite nested because some arrays are semantically arrays and other is dictionaries?

Reply:
Yes, I added tests to make it clearer what’s going on inside. I also added a test that checks the consistency between RESP2 and RESP3 responses.

Reply:
@alschastny If you don't mind, I think `arrayToDictionnary()` method could be enhanced so it will accept a list of keys that should be excluded from parsing, WDYT? I can jump on it later this week